### PR TITLE
fix Poetry dependency

### PR DESCRIPTION
**Pull request by dancju**
- Upgraded Gym to Gymnasium Since SB3 will only support Gymnasium in the future. Some environment classes were upgraded because the API for reset() and step() [was changed](https://gymnasium.farama.org/content/migration-guide/). We might want to update other environment classes as well.
- Upgraded Python to 3.10 in order to fix the dependency issue.
- Fixed minor bugs in Stock_NeurIPS2018_3_Backtest.ipynb and plot the backtesting results of all models (A2C, DDPG, PPO, TD3, SAC).

### CODE REVIEW

**Comment by zhumingpassional** [comment]
what is the meaning of  alpaca-trade-api = "^3"?

**Comment by dancju** [comment]
- "^3" is equivalent to ">=3.0.0,<4.0.0"
https://python-poetry.org/docs/dependency-specification/
- Gym is no longer maintained by OpenAI. Other projects in the OpenAI universe will only support Gymnasium. Gymnasium also has a new vector API. With vector API, you can train a single-instrument-trading agent on multiple financial instruments simultaneously with multithreading. RL training would be much much faster.
https://github.com/Farama-Foundation/Gymnasium/issues/12
- Even though SB3 has its own Vector API, SB3 will only support Gymnasium in the future. Currently, SB3 1.8.0 depends on Gym, and SB3 2.0.0a5 depends on Gymnasium.

**Comment by zhumingpassional** [comment]
ok. thanks very much for your codes and response.

**Comment by zhumingpassional** [comment]
yes. I see

File: pyproject.toml [hunk 1]
```diff
@@ -24,31 +25,27 @@ keywords=["Reinforcement Learning", "Finance"]
...
-tensorboardX = "*"
-yfinance = "*"
-stockstats = ">=0.4.0"
-scikit-learn = ">=0.21.0"
-ray = {extras=["default", "tune"], version=">=2.0.0"}
-stable-baselines3 = "^1.6.2"
-wrds = ">=3.1.6"
+alpaca-trade-api = "^3"
+ccxt = "^3"
+exchange-calendars = "^4"
```
Comment (HenryYYang) [inline pyproject.toml:32]: I think this cause a regression fixed by https://github.com/AI4Finance-Foundation/FinRL/pull/695

**Pull request merged**